bundles/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpBinding.java

```diff
@@ -303,8 +308,31 @@ private void formatAndExecute(String itemName, Command command, Type value) {
             url = String.format(url, Calendar.getInstance().getTime(), value);
         }
 
-        if (isNotBlank(httpMethod) && isNotBlank(url)) {
-            HttpUtil.executeUrl(httpMethod, url, provider.getHttpHeaders(itemName, command), null, null, timeout);
+        String body = provider.getBody(itemName, command);
+        InputStream stream = null;
+
+        if (StringUtils.isNotBlank(httpMethod) && StringUtils.isNotBlank(url)) {
+            if (httpMethod.equals("POST") && StringUtils.isNotBlank(body)) {
+                try {
+                    stream = IOUtils.toInputStream(body, "UTF-8");
+                } catch (IOException ioe) {
+                    logger.error("Failed to convert the specified body into an acceptable input stream.", ioe);
+                    logger.error("Body contents: {}", body);
+                }
+                logger.debug("Executing url '{}' via method {}, with body content '{}'", url, httpMethod, body);
+                HttpUtil.executeUrl(httpMethod, url, provider.getHttpHeaders(itemName, command), stream, "text/plain",
```

Comment:
I agree it's very limiting to fit everything sensibly inside a binding config string.  In the case where someone needs a different content type, the answer should be to use the [HTTP action](https://github.com/openhab/openhab/wiki/Actions#http-actions) triggered from a rule.  In fact, the workaround for the absence of this PR is:

.items
```
String PostBody
```

.rules
```
rule PostBodyToSomewhere
when
  Item PostBody received command
then
  val url = "http://server/path"
  sendHttpPostRequest(url, "text/plain", receivedCommand.toString)
end